tests/frontend/show_artifact_cas_digest.py

```diff
@@ -0,0 +1,291 @@
+
+    received_digest = result.output.splitlines()[0]
+    assert received_digest == "{target},{digest}".format(target=target, digest=expected_no_digest)
+
+
+# This tests that a target that hasn't been built locally
+# and that isn't cached remotely has no artifact CAS
+# digest.
+#
+# The test is performed with a remote cache.
+#
+@pytest.mark.datafiles(DATA_DIR)
+@pytest.mark.parametrize(
+    "target",
+    [
+        ("import-basic-files.bst"),
+        ("import-executable-files.bst"),
+        ("import-symlinks.bst"),
+    ],
+)
+def test_show_artifact_cas_digest_uncached(cli, tmpdir, datafiles, target):
+    project = str(datafiles)
+    expected_no_digest = ""
+
+    # Configure a local cache
+    local_cache = os.path.join(str(tmpdir), "cache")
+    cli.configure({"cachedir": local_cache})
+
+    with create_artifact_share(os.path.join(str(tmpdir), "artifactshare")) as share:
+
+        cli.configure({"artifacts": {"servers": [{"url": share.repo}]}})
+
+        # Check the target has not been built locally and is not existing in the remote cache
+        assert cli.get_element_state(project, target) == "buildable"
+        assert_not_shared(cli, share, project, target)
+
+        # Check the target has no artifact digest
+        result = cli.run(project=project, silent=True, args=["show", "--format", "%{name},%{artifact-cas-digest}", target])
+        result.assert_success()
+
+        received_digest = result.output.splitlines()[0]
+        assert received_digest == "{target},{digest}".format(target=target, digest=expected_no_digest)
+
+
+# This tests that a target that has been built locally and
+# that isn't cached remotely has an artifact CAS digest.
+#
+# The test is performed with a remote cache.
+#
+@pytest.mark.datafiles(DATA_DIR)
+@pytest.mark.parametrize(
+    "target, expected_digest",
+    [
+        ("import-basic-files.bst", "7093d3c89029932ce1518bd2192e1d3cf60fd88e356b39195d10b87b598c78f0/168"),
+        ("import-executable-files.bst", "133a9ae2eda30945a363272ac14bb2c8a941770b5a37c2847c99934f2972ce4f/170"),
+        ("import-symlinks.bst", "95947ea55021e26cec4fd4f9de90d2b7f4f7d803ccc91656b3e1f2c9923ddf19/131"),
+    ],
+)
+def test_show_artifact_cas_digest_cached_locally(cli, tmpdir, datafiles, target, expected_digest):
+    project = str(datafiles)
+
+    # Configure a local cache
+    local_cache = os.path.join(str(tmpdir), "cache")
+    cli.configure({"cachedir": local_cache})
+
+    with create_artifact_share(os.path.join(str(tmpdir), "artifactshare")) as share:
+
+        cli.configure({"artifacts": {"servers": [{"url": share.repo}]}})
+
+        # Build the target locally
+        result = cli.run(project=project, silent=True, args=["build", target])
+        result.assert_success()
+
+        # Check the target has been built locally and is not existing in the remote cache
+        assert cli.get_element_state(project, target) == "cached"
+        assert_not_shared(cli, share, project, target)
```

Comment:
I don't understand why/how this can be true, did you do something explicitly to make sure that building would not push the artifact to the remote ?

Reply:
In my tests I don't remember the built artifacts to be automatically pushed to the cache, that's why I push it manually in other tests and why I check the artifacts aren't cached remotely in some other. Is there a parameter I should set to enable automatically sharing the artifacts?

Reply:
Fixed by adding `"push": True` as discussed in our meeting.